src/aioquic/tls.py

```diff
@@ -365,7 +369,9 @@ def pull_block(buf: Buffer, capacity: int) -> Generator:
     length = int.from_bytes(buf.pull_bytes(capacity), byteorder="big")
     end = buf.tell() + length
     yield length
-    assert buf.tell() == end
+    if buf.tell() != end:
+        # There was trailing garbage or our parsing was bad.
+        raise AlertDecodeError("extra bytes at the end of a block")
```

Comment:
I think this should remain an assertion. It means the caller of `pull_block` did not read the right number of bytes within the context manager.

Reply:
No scratch that comment.